pkg/helm/helm.go

```diff
@@ -66,13 +68,24 @@ func (h *MetalLBChart) GetObjects(crdConfig *metallbv1beta1.MetalLB, withPrometh
 	if err != nil {
 		return nil, err
 	}
-	for _, obj := range objs {
+	for i, obj := range objs {
 		// Set namespace explicitly into non cluster-scoped resource because helm doesn't
 		// patch namespace into manifests at client.Run.
 		objKind := obj.GetKind()
 		if objKind != "PodSecurityPolicy" {
 			obj.SetNamespace(h.namespace)
 		}
+		// patch affinity and resources parameters explicitly into appropriate obj.
+		// This is needed because helm template doesn't support loading non table
+		// structure values.
+		objs[i], err = overrideControllerParameters(crdConfig, objs[i])
```

Comment:
I'd use obj as the arg.

Reply:
this somehow doesn't work with unstructured.Unstructured ptrs on override methods, so had to use index.